csep/core/poisson_evaluations.py

```diff
@@ -179,7 +180,12 @@ def conditional_likelihood_test(gridded_forecast, observed_catalog, num_simulati
         _ = observed_catalog.region.magnitudes
     except CSEPCatalogException:
         observed_catalog.region = gridded_forecast.region
-    gridded_catalog_data = observed_catalog.spatial_magnitude_counts()
+
+    if isinstance(gridded_forecast.region, QuadtreeGrid2D): #-----Asim's addition----
+        print('Quadtree grid detected.... ')
+        gridded_catalog_data = gridded_forecast.region.get_spatial_magnitude_counts(observed_catalog)
```

Comment:
Why are we using a different call here? From looking at this it seems to defeat the purpose of providing different region objects. I think this method should be renamed `spatial_magnitude_counts` like the other region classes.

Reply:
Thank you for pointing this out. Actually, I wrote this comment for myself temporary,  when I work in editor, I tend to lose sight of it.

As I recall, I had a good reason for calling that separately. You had the following line for for this:

> gridded_catalog_data = observed_catalog.spatial_magnitude_counts()

Your function for finding in class Catalog uses uses _bin_catalog_spatio_magnitude_counts() function, which in trun uses bin1d_vec(). So these functions are not available for quadtree class. So I had to write my own function in Quadtree class and then had to call this function through  region class like: observed_catalog.region.___ . So I had to call it separately.

I think if you can somehow route this function to region class, then I would not have to do this. 
I hope you understand my point.

Reply:
yes i do, that is a good point. i think we decided before that we would re-write this function using the `Region` class method `get_index_of` and replace the current implementation in the Catalog class. I will work on this in the next couple of days. If I can, I will commit these changes directly to this PR. 

That way we will be able to use any region class that defines `get_index_of` to compute the spatial counts and spatial magnitude counts.